crates/ty_python_semantic/src/types.rs

```diff
@@ -7492,6 +7531,42 @@ impl<'db> TypeAliasType<'db> {
     }
 }
 
+#[salsa::interned(debug)]
+pub struct BareTypeAliasType<'db> {
+    #[returns(ref)]
+    pub name: String,
```

Comment:
nit: I think `ast::name::Name` would probably be slightly more efficient, as `Name` wraps a `CompactString`, and the names of type aliases are likely to nearly always be pretty small strings

```suggestion
    pub name: ast::name::Name,
```